openff/models/types.py

```diff
@@ -133,7 +133,7 @@ def json_loader(data: str) -> dict:
             # Directly look for an encoded FloatQuantity/ArrayQuantity,
             # which is itself a dict
             v = json.loads(val)
-        except json.JSONDecodeError:
+        except (json.JSONDecodeError, TypeError):
```

Comment:
Would this also cover stuff like floats, strings, and primitive lists? If so, could we test add those to the test?

Reply:
Yes, tested that locally 
![image](https://user-images.githubusercontent.com/11765982/190681022-db5f2712-f708-4e97-b9da-730b7ff14c33.png)
but was too lazy to add to the tests, will do!